### PR TITLE
[rpc/distributed] eliminate code duplication in distributed/rendezvou…

### LINKED ISSUE #74440: [BE] Consolidate duplicated rendezvous code
### 🐛 Describe the bug

As we introduce dynamic group in #73372, the two rendezvous paths have some duplicated code. 

https://github.com/pytorch/pytorch/blob/f0db315efeef27a2d8760cec4573bcbe40854b01/torch/distributed/rendezvous.py#L112-L121

https://github.com/pytorch/pytorch/blob/f0db315efeef27a2d8760cec4573bcbe40854b01/torch/distributed/rendezvous.py#L85-L94

### Versions

 #73372

### Note

It will be easier to consolidate the duplicated rendezvous code once rank is not required for initialization of dynamic RPC members.

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

### PR DESCRIPTION
**Pull request by billdarrow**
This change eliminates duplication in redundant code paths inside of distributed/rendezvous.py.  Minor additional test coverage is added.  

Fixes #74440

### CODE REVIEW

File: torch/distributed/rendezvous.py [hunk 1]
```diff
@@ -50,35 +50,34 @@ def register_rendezvous_handler(scheme, handler):
...
             "rank" not in query_dict and "world_size" not in query_dict
         ), "The url: {url} has node-specific arguments(rank, world_size) already.".format(
             url=url
         )
         if rank != -1:
-            query_dict["rank"] = rank
-        if world_size != -1:
-            query_dict["world_size"] = world_size
-
+            query_dict["rank"] = str(rank)
```
Comment (H-Huang) [inline torch/distributed/rendezvous.py:78]: nit: I don't think rank and world_size need to be casted to strings

**Review by H-Huang — APPROVED** [review]
LGTM. Thanks for refactoring this code and making it a lot nicer. Left a minor comment.

File: torch/distributed/rendezvous.py (hunk 1 above)
Comment (billdarrow) [reply torch/distributed/rendezvous.py]: updated.

**Review by H-Huang — APPROVED** [review]
Thanks!

File: torch/distributed/rendezvous.py [hunk 2]
```diff
@@ -76,9 +76,8 @@ def rendezvous(url: str, rank: int = -1, world_size: int = -1, **kwargs):
         )
         if rank != -1:
             query_dict["rank"] = rank
```
Comment (H-Huang) [inline torch/distributed/rendezvous.py:78]: @billdarrow My bad, it looks like _query_to_dict returns Dict[str,str]. So these arguments do actually have to be string to pass the linter. We can either update the python type hints or the arguments.

Comment (billdarrow) [reply torch/distributed/rendezvous.py]: No problem, I added back the str() casts...

**Pull request closed without merging**